Common/src/grid_movement_structure.cpp

```diff
@@ -608,6 +609,136 @@ void CVolumetricMovement::ComputeDeforming_Wall_Distance(CGeometry *geometry, CC
   
 }
 
+void CVolumetricMovement::ComputeSolid_Wall_Distance(CGeometry *geometry, CConfig *config, su2double &MinDistance, su2double &MaxDistance) {
+
+  su2double *coord, dist2, dist;
+  unsigned short iDim, iMarker;
+  unsigned long iPoint, iVertex, nVertex_SolidWall;
+
+  int rank = MASTER_NODE;
+  int iProcessor, nProcessor = SINGLE_NODE;
+#ifdef HAVE_MPI
+  MPI_Comm_rank(MPI_COMM_WORLD, &rank);
+  MPI_Comm_size(MPI_COMM_WORLD, &nProcessor);
+#endif
+
+  MaxDistance = -1E22; MinDistance = 1E22;
+
+  if (rank == MASTER_NODE)
+    cout << "Computing distances to the nearest solid surface (except internal surfaces)." << endl;
+
+  unsigned long nLocalVertex_SolidWall = 0, nGlobalVertex_SolidWall = 0, MaxLocalVertex_SolidWall = 0;
+  unsigned long *Buffer_Send_nVertex    = new unsigned long [1];
+  unsigned long *Buffer_Receive_nVertex = new unsigned long [nProcessor];
+
+  /*--- Count the total number of nodes on deforming boundaries within the
+   local partition. ---*/
+
+  for (iMarker = 0; iMarker < config->GetnMarker_All(); iMarker++)
+    if (config->GetMarker_All_KindBC(iMarker) != SEND_RECEIVE &&
```

Comment:
I have changed SOLID_WALL_DISTANCE to BOUNDARY_DISTANCE in the C++ implementation but not in the config file option to keep the compatibility with previous versions.